app/controlplane/pkg/data/workflow.go

```diff
@@ -223,6 +225,22 @@ func applyWorkflowFilters(wfQuery *ent.WorkflowQuery, opts *biz.WorkflowListOpts
 			wfQuery = wfQuery.Where(workflow.HasProjectWith(project.NameIn(opts.WorkflowProjectNames...)))
 		}
 
+		// Append the JSON Filters to the query
+		if len(opts.JSONFilters) != 0 {
+			wfQuery = wfQuery.Where(func(selector *sql.Selector) {
+				// Build the predicates for each JSON filter
+				predicates := make([]*sql.Predicate, 0, len(opts.JSONFilters))
+				for _, filter := range opts.JSONFilters {
+					// Include the column where the filter is applied
+					filter.Column = workflow.FieldMetadata
+					jsonPredicate, _ := jsonfilter.BuildEntSelectorFromJSONFilter(filter)
+					predicates = append(predicates, jsonPredicate)
+				}
+				// Combine the predicates using OR logic
+				selector.Where(sql.Or(predicates...))
```

Comment:
interestingly, you chose to make `or` the default; it feels counterintuitive, no?

I mean if you provide multiple filters, my fist thought would be to become an `AND`. What do you think?

Reply:
You might be right I was just biassed by another PR. I've changed it to merge all filters in an AND. Thanks.